### ISSUE TITLE
Recent libgcrypt incompatibility

### ISSUE THREAD

**Issue by programmador** [gbevan/portage-brackets-bin-overlay#4]
There's a brackets-bin 1.7 package in hnaparst overlay which depends strictly on libgcrypt version 1.5.6-r101. This is right because trying to install brackets from your overlay causes it to live in the system with recent libgcrypt (currently 1.7.6), and brackets tells that it can't locate libgcrypt.so.11. Symlinking *.11 to *.20 does not make any sense because this way brackets still doesn't start because of abi incompatibility.

So I think more strict libgcrypt versioning should be used in ebuild dependencies. BTW it causes another problem: libgcrypt versions 1.7.6 and 1.5.6 have the same file /usr/lib64/libgcrypt.la file.
As for me - I resolved this conflict by keeping 1.7.6 version of libgcrypt.la file in /usr/lib64 and backuping 1.5.6 version in another folder. Anyway it can be resolved only manually.

I hope next versions of brackets binaries will be linked around newer libgcrypt. This will obviously need update of ebuild dependencies to a newer version too.

**Comment by programmador** [comment]
Oh, sorry, looks like I've created another duplicate of #1. Didn't search for closed issues first.
The libgcrypt-compat package was proposed there but there's no such package in Gentoo. Gentoo uses subslots for this case. If I understood correctly this simply means that two versions of the same package should be installed (which I have actually done in my system). Also it means that anyway dependency version should be stricted to 1.5.* in ebuild.
But.... Also if I understand correctly this will brake compatibility with funtoo :-)
So I don't actually know what should be done with ebuilds.

**Comment by gbevan** [comment]
Hi @programmador, I've been running brackets 1.8 with libgcrypt 1.7.3 ok.  Although I recently migrated to brackets-electron (prereleases for the new 1.9).

I've just added the latest brackets 1.9 release to the overlay - it's still keyword ~amd64, so if you want to try it you will need to add it to /etc/portage/package.keywords.

**Comment by programmador** [comment]
$ LANG=C eix brackets-bin
[I] app-editors/brackets-bin [1]
     Available versions:  [m](~)1.6 [m]1.7 [m](~)1.8 (~)1.9 {chromium}
     Installed versions:  1.9(01:37:54 03/26/17)(chromium)
     Homepage:            http://brackets.io
     Description:         open source code editor for web designers and front-end developers

[1] "gbevan-brackets" /var/lib/layman/gbevan-brackets


$ LANG=C eix dev-libs/libgcrypt
[I] dev-libs/libgcrypt
     Available versions:  
     (11)   (~)1.5.6-r101(11/11)
     (0)    1.7.3(0/20) ~1.7.5-r2(0/20) 1.7.6(0/20)
       {doc static-libs ABI_MIPS="n32 n64 o32" ABI_PPC="32 64" ABI_S390="32 64" ABI_X86="32 64 x32"}
     Installed versions:  1.7.6(19:36:45 03/22/17)(-doc -static-libs ABI_MIPS="-n32 -n64 -o32" ABI_PPC="-32 -64" ABI_S390="-32 -64" ABI_X86="64 -32 -x32")
     Homepage:            http://www.gnupg.org/
     Description:         General purpose crypto library based on the code used in GnuPG


$ brackets
/usr/bin/brackets: error while loading shared libraries: libgcrypt.so.11: cannot open shared object file: No such file or directory

Are you sure that you have only libgcrypt-1.7.* (and, as consequence, /usr/lib/libgcrypt.so.11 missing) and everything works fine?

**Comment by programmador** [comment]
Actually it shouldn't work:
https://github.com/adobe/brackets/issues/10255

**Comment by gbevan** [comment]
```
$ eix gcrypt
[I] dev-libs/libgcrypt
     Available versions:  
     (11)   ~1.5.6-r101(11/11)
     (0)    1.7.3(0/20) ~1.7.5-r2(0/20) ~1.7.6(0/20)
       {doc static-libs ABI_MIPS="n32 n64 o32" ABI_PPC="32 64" ABI_S390="32 64" ABI_X86="32 64 x32"}
     Installed versions:  1.7.3(19:56:58 23/08/16)(-doc -static-libs ABI_MIPS="-n32 -n64 -o32" ABI_PPC="-32 -64" ABI_S390="-32 -64" ABI_X86="32 64 -x32")
     Homepage:            http://www.gnupg.org/
     Description:         General purpose crypto library based on the code used in GnuPG

$ ll /usr/lib/libgcrypt.so*
lrwxrwxrwx 1 root root      19 Aug 23  2016 /usr/lib/libgcrypt.so -> libgcrypt.so.20.1.3
lrwxrwxrwx 1 root root      19 Apr 10  2015 /usr/lib/libgcrypt.so.11 -> libgcrypt.so.11.8.3
-rwxr-xr-x 1 root root  524576 Apr 10  2015 /usr/lib/libgcrypt.so.11.8.3
lrwxrwxrwx 1 root root      19 Aug 23  2016 /usr/lib/libgcrypt.so.20 -> libgcrypt.so.20.1.3
-rwxr-xr-x 1 root root 1109672 Aug 23  2016 /usr/lib/libgcrypt.so.20.1.3
$ equery -C belongs /usr/lib/libgcrypt.so.11
 * Searching for /usr/lib/libgcrypt.so.11 ... 
dev-libs/libgcrypt-1.7.3 (/usr/lib64/libgcrypt.so.11.8.3)
```
Perhaps libgcrypt.so.11 was dropped in 1.7.6.

**Comment by programmador** [comment]
Hm... Maybe.
A manual installation of 1.5* does the trick if 1.7.6 installed:
```
$ emerge --ask =dev-libs/libgcrypt-1.5.6-r101
$ equery -C belongs /usr/lib/libgcrypt.so.11
 * Searching for /usr/lib/libgcrypt.so.11 ... 
dev-libs/libgcrypt-1.5.6-r101 (/usr/lib64/libgcrypt.so.11.8.5)
```
But as I mentioned while starting the issue, a file conflict will occur (/usr/lib64/libgcrypt.la). I backed up both versions:
1. Moved (but not copied) 1.7.6 version to backup.
2. Installed 1.5.6.
3. Moved (but not copied) 1.5.6 version to backup
4. Copied 1.7.6 version from backup to /usr/lib64

**Comment by gbevan** [comment]
I've tried forcing slot 11 (1.5.6) in the build, but the install of 1.5.6 has collisions with 1.7.6.  So looks like a no-go.

I'd recommend trying brackets-electron - https://github.com/zaggino/brackets-electron.  I've been using this recently and it seems more stable and faster to me.  I've just tested the build and run are ok without libgcrypt.so.11.

I'll probably drop support for this ebuild going fwd...
